src/IniFile.cpp

```diff
@@ -49,6 +49,9 @@ cIniFile::cIniFile(void) :
 
 bool cIniFile::ReadFile(const AString & a_FileName, bool a_AllowExampleRedirect)
 {
+
+	m_Filename = a_FileName;
+
```

Comment:
I still don't like this and would prefer if you changed the pointer to shared and let the files be written explicitly. Because now there are two functions that dump the contents to a file, and their interaction is not exactly clear. If you call WriteFile(), will it overwrite the filename stored for Flush()?